### DISCUSSION TITLE
Why is it necessary to define the terminal tasks of a pipeline?

### DISCUSSION THREAD

**Discussion by marip8** [tesseract-robotics/tesseract_planning#365] [Q&A] [answered]
Are the terminal tasks not implicitly all the leaf nodes of the task graph? If so, we should be able to identify those without user input

https://github.com/tesseract-robotics/tesseract_planning/blob/4e7f323962ce1e3e3b1b7f903fbd5d7041c69c36/tesseract_task_composer/config/task_composer_plugins.yaml#L65

**Comment by Levi-Armstrong** [comment, marked as answer]
They are the leaf nodes but the order is important to identify what the return value should be.
